spacy/training/loggers.py

```diff
@@ -1,18 +1,22 @@
-from typing import Dict, Any, Tuple, Callable, List
+from typing import Dict, Any, Tuple, Callable, List, Optional
+import wasabi
+import tqdm
 
 from ..util import registry
 from .. import util
 from ..errors import Errors
-from wasabi import msg
 
 
 @registry.loggers("spacy.ConsoleLogger.v1")
-def console_logger():
+def console_logger(progress_bar: bool=False):
     def setup_printer(
         nlp: "Language",
+        printer: Optional[wasabi.Printer]=wasabi.Printer()
```

Comment:
Just a stylistic preference but I would probably do `from wasabi import Printer` etc.

Reply:
I just figured it'd be nice if people are looking at the signature line

Reply:
It's kind of tricky to use a type from some other library like this, so I leaned towards explicitness

Reply:
Now that I'm looking at the implementation again, I'm not sure I like having the `wasabi` printer detail leak into the API like this. I previously had it like this in an earlier version and then opted for making the _behaviour_ more explicit with `silent`, because that's what we're actually trying to solve here: we need a way to tell the function whether it's running in terminal/"pretty" context or not.

For a user implementing a custom logger, having the `Printer` object passed in like this from the CLI has absolutely no significance. It also doesn't help with defining custom behaviours depending on the CLI vs. non-CLI context.